rlog/2024-12-23-de-mls.mdx

```diff
@@ -0,0 +1,252 @@
+
+de-MLS provides two different user management configurations, both utilizing the Waku protocol for DS:
+
+1. **Single Admin Over a Single Waku Node**: 
+   - A single authorized identity (admin) manages the group including removing or adding users.
+   
+2. **Multi-Admin Over Multiple Waku Nodes**: 
+   - Multiple admins have equal authority to add or remove users.
+    - A consensus mechanism ensures consistency by resolving concurrent changes within the same epoch and preventing possible conflicts. In each epoch, all modifications are managed exclusively by a single admin.
+
+In multi-admin settings, de-MLS requires a consensus among admins that have equal rights in the group since changes in an epoch in MLS are required to be conducted by a single identity, that is the admin. 
+
+For the consensus integration, ongoing research explores two promising approaches:
+
+1. **On-chain consensus mechanisms**: Outsourcing consensus to a smart contract solution for transparent and immutable agreement.
+2. **Off-chain consensus mechanisms**: Utilizing off-chain consensus protocol to design efficient, decentralized protocols.
+
+### Waku Integration
+
+Waku integration is one of step of building the de-MLS by replacing the client-server messages with the Waku communication. The main challenge in moving to a decentralized system was converting the centralized DS to the decentralized DS; which has two key functions.
+
+1. Sending out messages, which includes not only sending to the correct user but also deciding on the correct sequence (docs reference here). 
+2. Manages the key package of users. 
+
+To ensure decentralization, key packages cannot be stored in a centralized location. Initially, the idea of using a smart contract (SC) as a replacement for the server-side, from the package manager's perspective, was considered. However, this approach proved unsuitable because, once data is added to the blockchain, it cannot be deleted without leaving a trace. One of the key requirements is that each package must be used only once and then deleted afterward, which conflicts with the immutable nature of blockchain storage. Instead, users must provide them upon request to be validated.
+
+### Distribution of the Messages
+
+Since the Waku is based on gossipsub protocol and accordingly from the network point of view joining a group for a user will look like subscribing to the necessary topics.
+
+We have pub-sub topics that send messages in the network. We also have content topics, which determine the groups and what messages are sent.
+
+There are three types of topics: 
+- `app_msg`: The owner of the topic where messages encrypted with the group key are sent, thanks to encryption, even if someone not from the group subscribes to the topic, they will not be able to read the correspondence itself.
```

Comment:
Not sure I understand the concept of "owner of topic"

Reply:
I didn't phrase it correctly, thanks for seeing that, I'll correct it now